### PR TITLE
[eas-cli] Add learn more link to project ID mismatch errors

### PR DESCRIPTION
**Pull request by brentvatne**
# Checklist

- [x] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary. You can comment this pull request with `/changelog-entry [breaking-change|new-feature|bug-fix|chore] [message]` and CHANGELOG.md will be updated automatically.

# Why

Error messages are frustrating, and by providing a link to an FYI page we can easily provide more context on what is happening, why, and suggestions for how to adapt your workflow.

# How

I created a simple placeholder FYI page and then linked to it from here. I also changed the formatting of the message slightly to inline the mismatched values in parens.

# Test Plan

I changed the slug and owner in an existing project and tried running commands.

image

### CODE REVIEW

**Comment by brentvatne** [comment]
/changelog-entry chore Improve message when project ID doesn't match either slug or owner

**Review by wkozyra95 — APPROVED** [review]

File: packages/eas-cli/src/commandUtils/context/contextUtils/getProjectIdAsync.ts [hunk 1]
```diff
@@ -79,13 +79,21 @@ export async function getProjectIdAsync(
     const appForProjectId = await AppQuery.byIdAsync(localProjectId);
     if (exp.owner && exp.owner !== appForProjectId.ownerAccount.name) {
       throw new Error(
-        `Project config: Project identified by 'extra.eas.projectId' is not owned by owner specified in the 'owner' field. (project = '${appForProjectId.ownerAccount.name}', config = '${exp.owner}')`
+        `Project config: Project identified by 'extra.eas.projectId' (${
+          appForProjectId.ownerAccount.name
+        }) is not owned by owner specified in the 'owner' field (${exp.owner}). ${learnMore(
+          'https://expo.fyi/eas-project-id'
+        )}`
```
Comment (dsokal) [inline packages/eas-cli/src/commandUtils/context/contextUtils/getProjectIdAsync.ts:86]: We use double quotes to reference paths from app.json:
- https://github.com/expo/eas-cli/blob/main/packages/eas-cli/src/project/ios/bundleIdentifier.ts#L93
- https://github.com/expo/eas-cli/blob/main/packages/eas-cli/src/project/android/applicationId.ts

```suggestion
        `Project config: Project identified by "extra.eas.projectId" (${
          appForProjectId.ownerAccount.name
        }) is not owned by owner specified in the "owner" field (${exp.owner}). ${learnMore(
          'https://expo.fyi/eas-project-id'
        )}`
```

File: packages/eas-cli/src/commandUtils/context/contextUtils/getProjectIdAsync.ts [hunk 2]
```diff
@@ -79,13 +79,21 @@ export async function getProjectIdAsync(
...
     }
 
     if (exp.slug && exp.slug !== appForProjectId.slug) {
       throw new Error(
-        `Project config: Slug for project identified by 'extra.eas.projectId' does not match the 'slug' field. (project = '${appForProjectId.slug}', config = '${exp.slug}')`
+        `Project config: Slug for project identified by 'extra.eas.projectId' (${
+          appForProjectId.slug
+        }) does not match the 'slug' field (${exp.slug}). ${learnMore(
+          'https://expo.fyi/eas-project-id'
+        )}`
```
Comment (dsokal) [inline packages/eas-cli/src/commandUtils/context/contextUtils/getProjectIdAsync.ts:96]:
```suggestion
        `Project config: Slug for project identified by "extra.eas.projectId" (${
          appForProjectId.slug
        }) does not match the "slug" field (${exp.slug}). ${learnMore(
          'https://expo.fyi/eas-project-id'
        )}`
```

File: packages/eas-cli/src/commandUtils/context/contextUtils/__tests__/getProjectIdAsync-test.ts [hunk 3]
```diff
@@ -96,7 +97,9 @@ describe(getProjectIdAsync, () => {
         { nonInteractive: false }
       )
     ).rejects.toThrow(
-      `Project config: Project identified by 'extra.eas.projectId' is not owned by owner specified in the 'owner' field. (project = 'notnotbrent', config = 'wat')`
+      `Project config: Project identified by 'extra.eas.projectId' (notnotbrent) is not owned by owner specified in the 'owner' field (wat). ${learnMore(
```
Comment (dsokal) [inline packages/eas-cli/src/commandUtils/context/contextUtils/__tests__/getProjectIdAsync-test.ts:100]:
```suggestion
      `Project config: Project identified by "extra.eas.projectId" (notnotbrent) is not owned by owner specified in the "owner" field (wat). ${learnMore(
```

File: packages/eas-cli/src/commandUtils/context/contextUtils/__tests__/getProjectIdAsync-test.ts [hunk 4]
```diff
@@ -117,7 +120,9 @@ describe(getProjectIdAsync, () => {
         { nonInteractive: false }
       )
     ).rejects.toThrow(
-      `Project config: Slug for project identified by 'extra.eas.projectId' does not match the 'slug' field. (project = 'test', config = 'wat')`
+      `Project config: Slug for project identified by 'extra.eas.projectId' (test) does not match the 'slug' field (wat). ${learnMore(
```
Comment (dsokal) [inline packages/eas-cli/src/commandUtils/context/contextUtils/__tests__/getProjectIdAsync-test.ts:123]:
```suggestion
      `Project config: Slug for project identified by "extra.eas.projectId" (test) does not match the "slug" field (wat). ${learnMore(
```

**Review by dsokal — APPROVED** [review]

**Review by wschurman — APPROVED** [review]

**Pull request merged**